src/configfetcher.cpp

```diff
@@ -53,6 +55,67 @@ int ConfigFetcher::ProgressFunction(curl_off_t dltotal, curl_off_t dlnow, curl_o
     return closed ? 1 : 0;  // Return 0 to continue, or 1 to abort
 }
 
+std::string getCertificateName(X509* const Certificate)
+{
+    char StaticBuffer[2048];
+    X509_NAME_oneline(X509_get_subject_name(Certificate), StaticBuffer, sizeof(StaticBuffer));
+    return std::string(StaticBuffer);
+}
+
+CURLcode SslContextCallback(CURL* curl, void* sslctx, void* raw_data)
+{
+    return static_cast<ConfigFetcher*>(raw_data)->SslContextSetup(curl, sslctx);
+}
+
+CURLcode ConfigFetcher::SslContextSetup(CURL* curl, void* sslctx) {
+    LOG_DEBUG << "Loading extra certificates";
+
+    if (sslctx == nullptr) {
+        LOG_ERROR(9999) << "Invalid callback arguments";
+        return CURLE_ABORTED_BY_CALLBACK;
+    }
+
+    X509_STORE* store = SSL_CTX_get_cert_store(static_cast<SSL_CTX*>(sslctx));
+
+    for (const auto & entryPath : extraSslCertificates) {
+        std::ifstream file(entryPath.c_str());
+        if(!file) {
+            LOG_WARN(9999) << "Couldn't read file: " << entryPath;
+            continue;
+        }
+
+        static const char beginCertificateString[] = "-----BEGIN CERTIFICATE-----";
+        static const char endCertificateString[] = "-----END CERTIFICATE-----";
+
+        std::stringstream fileContentStream;
+        fileContentStream << file.rdbuf();
+        std::string fileContent = fileContentStream.str();
+        const char* certBuffer = fileContent.c_str();
+
+        while (nullptr != (certBuffer = strstr(certBuffer, beginCertificateString))) {
+            const char* endString = strstr(certBuffer, endCertificateString);
+            if (endString != nullptr) {
+                size_t lengthOfCertificateData = endString - certBuffer + sizeof(endCertificateString);
+                BIO* certificateBio = BIO_new_mem_buf(const_cast<char*>(certBuffer), static_cast<int>(lengthOfCertificateData));
+                X509* certificate = PEM_read_bio_X509(certificateBio, nullptr, nullptr, nullptr);
+
+                const int status = X509_STORE_add_cert(store, certificate);
+                if (status == 0) {
+                    LOG_WARN(9999) << "Error adding certificate: " << getCertificateName(certificate);
```

Comment:
```suggestion
                    LOG_WARN(0) << "Error adding certificate: " << getCertificateName(certificate);
```

Could you please change the 9999 log_id-s to 0. We use 0 id for uncategorized log messages.